includes/openid-connect-generic-client-wrapper.php

```diff
@@ -371,33 +405,43 @@ function authentication_request_callback() {
 
 		// validate the found / created user
 		$valid = $this->validate_user( $user );
-		
-		if ( is_wp_error( $valid ) ){
-			$this->error_redirect( $valid );
+
+		if ( is_wp_error( $valid ) ) {
+			return $valid;
 		}
 
 		// login the found / created user
-		$this->login_user( $user, $token_response, $id_token_claim, $user_claim, $subject_identity  );
+		$this->login_user( $user, $token_response, $id_token_claim, $user_claim, $subject_identity );
 
 		do_action( 'openid-connect-generic-user-logged-in', $user );
 
 		// log our success
 		$this->logger->log( "Successful login for: {$user->user_login} ({$user->ID})", 'login-success' );
 
-		// redirect back to the origin page if enabled
-		$redirect_url = isset( $_COOKIE[ $this->cookie_redirect_key ] ) ? esc_url( $_COOKIE[ $this->cookie_redirect_key ] ) : false;
+		return $user;
+	}
 
-		if( $this->settings->redirect_user_back && !empty( $redirect_url ) ) {
-			do_action( 'openid-connect-generic-redirect-user-back', $redirect_url, $user );
-			setcookie( $this->cookie_redirect_key, $redirect_url, 1, COOKIEPATH, COOKIE_DOMAIN, is_ssl() );
-			wp_redirect( $redirect_url );
+	function authenticate_filter($user, $username, $password) {
+		if ( $user instanceof WP_User ) {
+			return $user;
 		}
-		// otherwise, go home!
-		else {
-			wp_redirect( home_url() );
+
+		if ( is_null($user) || is_wp_error($user) && $user->get_error_code() === 'invalid_username' ) {
```

Comment:
I need a little clarity on this check. Do we want to test for the exact message 'invalid_username' here?

Implication being that we only send a remote authentication request if the provided username doesn't exist in WP?

Reply:
You are right.
In current implementation (priority 15) makes this check no sense. Its make sense only if this filter runs after the default wordpress authenticator (priority 20). the default returns the WP_Error if a user not found in prefix_users.
Q: What would be better here?
1. to modify this code and remove this check.
2. to allow administrator to set this priority on settings page? (default 15)